src/datadoc_editor/frontend/callbacks/register_callbacks.py

```diff
@@ -526,3 +464,54 @@ def callback_accept_dataset_metadata_date_input(
             contains_data_from,
             contains_data_until,
         )
+
+    @app.callback(
+        Output({"type": "pseudo-field-container", "variable": MATCH}, "children"),
+        Input({"type": "pseudonymization-dropdown", "variable": MATCH}, "value"),
+        State({"type": "pseudonymization-dropdown", "variable": MATCH}, "id"),
+    )
+    def update_pseudo_fields(
+        selected_algorithm,  # noqa: ANN001
+        dropdown_id,  # noqa: ANN001
+    ) -> dbc.Form:
+        """Build editable pseudonymization fields dynamically based on selected pseudo algorithm."""
+        logger.debug("Selected algorithm: %s", selected_algorithm)
```

Comment:
```suggestion
        selected_algorithm = PseudonymizationAlgorithmsEnum[selected_algorithm]
        logger.debug("Selected algorithm: %s", selected_algorithm)
```

It would be nice to use the enum to have typed variables on all the functions this is passed in and out of. It's a bit hard to follow with the anonymous strings.